src/OpenTelemetry.Extensions/README.md

```diff
@@ -46,10 +46,35 @@ and adds the baggage keys and values to the `Activity` as tags (attributes) on s
 
 Add this activity processor to a tracer provider.
 
-Example of adding BaggageActivityProcessor to `TracerProvider`:
+For example, to add all baggage entries to new activities:
 
 ```cs
 var tracerProvider = Sdk.CreateTracerProviderBuilder()
-    .AddBaggageActivityProcessor()
+    .AddBaggageActivityProcessor(BaggageActivityProcessor.AllowAllBaggageKeys)
     .Build();
 ```
+
+Alternatively, you can select which baggage keys you want to copy using a
+custom predicate function.
+
+For example, to only copy baggage entries where the key start with `my-key`
+using a custom function:
+
+```cs
+var tracerProvider = Sdk.CreateTracerProviderBuilder()
+  .AddBaggageActivityProcessor((baggageKey) => baggageKey.StartWith("my-key", System.StringComparison.Ordinal))
+  .Build();
+```
+
+For example, to only copy baggage entries where the key matches the regular
+expression `^my-key`:
+
+```cs
+var baggageKeyRegex = new Regex("^mykey", RegexOptions.Compiled);
+var tracerProvider = Sdk.CreateTracerProviderBuilder()
+  .AddBaggageActivityProcessor((baggageKey) => baggageKeyRegex.IsMatch(baggageKey))
+  .Build();
+```
+
+Warning: The baggage key predicate is executed for every started activity.
```

Comment:
could you also add that it is executed for each entry in the Baggage.

Reply:
Update in [f147ef9](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1816/commits/f147ef9b1b9c803c48254a4302adc007e8ef92b4)